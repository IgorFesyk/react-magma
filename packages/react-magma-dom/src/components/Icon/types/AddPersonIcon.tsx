import * as React from 'react';
  import { IconProps, renderIcon } from '../utils';

  const iconType = {"viewBox":"0 0 35.21 28.01","paths":[{"d":"M36.29,4.74A.37.37,0,0,1,36.4,5h0v2.3a.4.4,0,0,1-.38.39H31.75V12a.4.4,0,0,1-.38.39h-2.3a.4.4,0,0,1-.38-.39V7.72H24.44a.39.39,0,0,1-.38-.38V4.94a.39.39,0,0,1,.38-.38h4.25V.37h0a.33.33,0,0,1,.1-.25A.38.38,0,0,1,29.06,0H31.4a.4.4,0,0,1,.27.12.38.38,0,0,1,.08.23V4.62H36a.45.45,0,0,1,.28.12Z","transform":"translate(-1.19 0.01)"},{"d":"M13.24,16a6,6,0,1,0-6-6A6,6,0,0,0,13.24,16Zm0,3c-4,0-12,2-12,6v3H25.27V25c0-4-8-6-12-6Z","transform":"translate(-1.19 0.01)"}]}
  
  export const AddPersonIcon: React.FunctionComponent<IconProps> = (props: IconProps) =>
    renderIcon(props, iconType);