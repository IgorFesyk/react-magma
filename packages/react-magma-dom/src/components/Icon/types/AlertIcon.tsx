import * as React from 'react';
  import { IconProps, renderIcon } from '../utils';

  const iconType = {"viewBox":"0 0 5.04 25.23","paths":[{"d":"M7.23,21.91h5v4.7h-5ZM7.41,1.39h4.71v7.5L10.88,20.15H8.66L7.41,8.89Z","transform":"translate(-7.23 -1.39)"}]}
  
  export const AlertIcon: React.FunctionComponent<IconProps> = (props: IconProps) =>
    renderIcon(props, iconType);