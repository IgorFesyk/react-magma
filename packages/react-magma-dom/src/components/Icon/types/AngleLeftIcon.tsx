import * as React from 'react';
import { IconProps, renderIcon } from '../utils';

const iconType = {
  viewBox: '0 0 9.09 15.59',
  paths: [
    {
      d:
        'M9.55,7.5a.48.48,0,0,1-.16.36L3.25,14l6.14,6.14a.46.46,0,0,1,.16.36.48.48,0,0,1-.16.36l-.78.78a.47.47,0,0,1-.71,0L.61,14.36A.51.51,0,0,1,.46,14a.49.49,0,0,1,.15-.36L7.9,6.36a.47.47,0,0,1,.71,0l.78.78A.46.46,0,0,1,9.55,7.5Z',
      transform: 'translate(-0.46 -6.2)'
    }
  ]
};

export const AngleLeftIcon: React.FunctionComponent<IconProps> = (
  props: IconProps
) => renderIcon(props, iconType);
