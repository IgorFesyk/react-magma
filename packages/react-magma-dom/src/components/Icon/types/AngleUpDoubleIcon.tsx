import * as React from 'react';
import { IconProps, renderIcon } from '../utils';

const iconType = {
  viewBox: '0 0 40 40',
  paths: [
    {
      d:
        'M19.9980925,18 C20.2931254,18 20.5544584,18.1138166 20.7820916,18.3414497 L36.6547352,34.2140934 C36.8823683,34.4417265 36.9961849,34.7030595 36.9961849,34.9980925 L37,34.9980925 C37,35.2931254 36.8861834,35.5544584 36.6585503,35.7820916 L34.9551167,37.4855251 C34.7274835,37.7131583 34.4661505,37.8269749 34.1711176,37.8269749 C33.8760847,37.8269749 33.6147516,37.7131583 33.3871185,37.4855251 L20,24.0984066 L6.61288151,37.4855251 C6.38524835,37.7131583 6.12391532,37.8269749 5.82888241,37.8269749 C5.53384949,37.8269749 5.27251646,37.7131583 5.0448833,37.4855251 L3.34144973,35.7820916 C3.11381658,35.5544584 3,35.2931254 3,34.9980925 C3,34.7030595 3.11381658,34.4417265 3.34144973,34.2140934 L19.2140934,18.3414497 C19.4417265,18.1138166 19.7030595,18 19.9980925,18 Z M19.9980925,2 C20.2931254,2 20.5544584,2.11381658 20.7820916,2.34144973 L36.6547352,18.2140934 C36.8823683,18.4417265 36.9961849,18.7030595 36.9961849,18.9980925 L37,18.9980925 C37,19.2931254 36.8861834,19.5544584 36.6585503,19.7820916 L34.9551167,21.4855251 C34.7274835,21.7131583 34.4661505,21.8269749 34.1711176,21.8269749 C33.8760847,21.8269749 33.6147516,21.7131583 33.3871185,21.4855251 L20,8.09840664 L6.61288151,21.4855251 C6.38524835,21.7131583 6.12391532,21.8269749 5.82888241,21.8269749 C5.53384949,21.8269749 5.27251646,21.7131583 5.0448833,21.4855251 L3.34144973,19.7820916 C3.11381658,19.5544584 3,19.2931254 3,18.9980925 C3,18.7030595 3.11381658,18.4417265 3.34144973,18.2140934 L19.2140934,2.34144973 C19.4417265,2.11381658 19.7030595,2 19.9980925,2 Z'
    }
  ]
};

export const AngleUpDoubleIcon: React.FunctionComponent<IconProps> = (
  props: IconProps
) => renderIcon(props, iconType);
