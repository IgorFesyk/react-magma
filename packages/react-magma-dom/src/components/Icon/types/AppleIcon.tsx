import * as React from 'react';
  import { IconProps, renderIcon } from '../utils';

  const iconType = {"viewBox":"0 0 22.8 28","paths":[{"d":"M22.29,14.84a6.37,6.37,0,0,1,3-5.33,6.54,6.54,0,0,0-5.12-2.77C18,6.52,15.94,8,14.83,8S12,6.77,10.21,6.8a6.89,6.89,0,0,0-5.79,3.51C2,14.6,3.79,21,6.2,24.42,7.37,26.12,8.77,28,10.61,28s2.45-1.14,4.59-1.14S18,28,19.82,27.93s3.12-1.73,4.29-3.44a15.84,15.84,0,0,0,1.94-4,6.18,6.18,0,0,1-3.76-5.67ZM18.76,4.44A6.17,6.17,0,0,0,20.22,0,6.24,6.24,0,0,0,16.1,2.09a5.86,5.86,0,0,0-1.49,4.33,5.19,5.19,0,0,0,4.15-2Z","transform":"translate(-3.25 0.03)"}]}
  
  export const AppleIcon: React.FunctionComponent<IconProps> = (props: IconProps) =>
    renderIcon(props, iconType);