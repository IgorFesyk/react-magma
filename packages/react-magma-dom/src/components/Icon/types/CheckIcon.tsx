import * as React from 'react';
import { IconProps, renderIcon } from '../utils';

const iconType = {
  viewBox: '0 0 29.37 21.88',
  paths: [
    {
      d: 'M26.71,3.06,11.89,17.88,4.4,10.4.87,13.93l11,11L30.24,6.59Z',
      transform: 'translate(-0.87 -3.06)'
    }
  ]
};

export const CheckIcon: React.FunctionComponent<IconProps> = (
  props: IconProps
) => renderIcon(props, iconType);
