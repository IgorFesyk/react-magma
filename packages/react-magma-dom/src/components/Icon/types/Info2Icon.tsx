import * as React from 'react';
import { IconProps, renderIcon } from '../utils';

const iconType = {
  viewBox: '0 0 27.3 27.3',
  paths: [
    {
      d:
        'M12.65,9.9V7.14h2.7V9.9ZM14,25A11,11,0,1,0,3,14,11,11,0,0,0,14,25ZM14,.35A13.65,13.65,0,1,1,.35,14,13.62,13.62,0,0,1,14,.35ZM12.65,20.86V12.65h2.7v8.21Z',
      transform: 'translate(-0.35 -0.35)'
    }
  ]
};

export const Info2Icon: React.FunctionComponent<IconProps> = (
  props: IconProps
) => renderIcon(props, iconType);
