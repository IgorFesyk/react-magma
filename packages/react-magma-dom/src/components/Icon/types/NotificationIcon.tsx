import * as React from 'react';
import { IconProps, renderIcon } from '../utils';

const iconType = {
  viewBox: '0 0 40 40',
  paths: [
    {
      d:
        'M20,4.5625 C15.877,4.5625 11.9998125,6.168 9.0845,9.0845 C6.1691875,12.001 4.5625,15.877 4.5625,20 C4.5625,24.123 6.168,28.0001875 9.0845,30.9155 C12.001,33.8308125 15.877,35.4375 20,35.4375 C24.123,35.4375 28.0001875,33.832 30.9155,30.9155 C33.8308125,27.999 35.4375,24.123 35.4375,20 C35.4375,15.877 33.832,11.9998125 30.9155,9.0845 C27.999,6.1691875 24.123,4.5625 20,4.5625 Z M20,1 L20,1 C30.4939375,1 39,9.5060625 39,20 C39,30.4939375 30.4939375,39 20,39 C9.5060625,39 1,30.4939375 1,20 C1,9.5060625 9.5060625,1 20,1 Z M17.625,27.125 L22.375,27.125 L22.375,31.875 L17.625,31.875 L17.625,27.125 Z M17.625,8.125 L22.375,8.125 L22.375,22.375 L17.625,22.375 L17.625,8.125 Z'
    }
  ]
};

export const NotificationIcon: React.FunctionComponent<IconProps> = (
  props: IconProps
) => renderIcon(props, iconType);
