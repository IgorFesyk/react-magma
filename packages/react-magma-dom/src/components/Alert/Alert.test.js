import React from 'react';
import { axe } from 'jest-axe';
import { Alert, AlertVariant } from '.';
import { render, fireEvent } from 'react-testing-library';
import { Info2Icon } from '../Icon/types/Info2Icon';
import { CheckIcon } from '../Icon/types/CheckIcon';
import { NotificationIcon } from '../Icon/types/NotificationIcon';
import { BlockedIcon } from '../Icon/types/BlockedIcon';

const MOCK_ID = 'abc123';

jest.mock('uuid', () => {
  return {
    v4: jest.fn(() => MOCK_ID)
  };
});

const alertText = 'Test Alert Text';

const renderAlert = (myProps = {}) => {
  return render(<Alert {...myProps}>{alertText}</Alert>);
};

describe('Alert', () => {
  it('should find element by testId', () => {
    const testId = 'test-id';
    const { getByTestId } = renderAlert({ testId });

    expect(getByTestId(testId)).toBeInTheDocument();
  });

  it('should render an alert with default variant', () => {
    const { container } = renderAlert();

    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).toHaveStyleRule('background-color', '#575757');

    const alertIcon = container.querySelector('svg');
    const { container: iconContainer } = render(
      <Info2Icon size={20} id={alertIcon.childNodes[0].id} />
    );
    const expectedIcon = iconContainer.querySelector('svg');

    expect(alertIcon).toEqual(expectedIcon);
  });

  describe('Variants', () => {
    it('should render an alert with info variant', () => {
      const { container } = renderAlert({ variant: AlertVariant.info });

      expect(container.firstChild).toHaveStyleRule(
        'background-color',
        '#575757'
      );

      const alertIcon = container.querySelector('svg');
      const { container: iconContainer } = render(
        <Info2Icon size={20} id={alertIcon.childNodes[0].id} />
      );
      const expectedIcon = iconContainer.querySelector('svg');

      expect(alertIcon).toEqual(expectedIcon);
    });

    it('should render an alert with success variant', () => {
      const { container } = renderAlert({ variant: AlertVariant.success });

      expect(container.firstChild).toHaveStyleRule(
        'background-color',
        '#3A8200'
      );

      const alertIcon = container.querySelector('svg');
      const { container: iconContainer } = render(
        <CheckIcon size={20} id={alertIcon.childNodes[0].id} />
      );
      const expectedIcon = iconContainer.querySelector('svg');

      expect(alertIcon).toEqual(expectedIcon);
    });

    it('should render an alert with warning variant', () => {
      const { container } = renderAlert({ variant: AlertVariant.warning });

      expect(container.firstChild).toHaveStyleRule(
        'background-color',
        '#FFC72C'
      );

      const alertIcon = container.querySelector('svg');
      const { container: iconContainer } = render(
        <NotificationIcon size={20} id={alertIcon.childNodes[0].id} />
      );
      const expectedIcon = iconContainer.querySelector('svg');

      expect(alertIcon).toEqual(expectedIcon);
    });

    it('should render an alert with danger variant', () => {
      const { container } = renderAlert({ variant: AlertVariant.danger });

      expect(container.firstChild).toHaveStyleRule(
        'background-color',
        '#E70000'
      );

      const alertIcon = container.querySelector('svg');
      const { container: iconContainer } = render(
        <BlockedIcon size={20} id={alertIcon.childNodes[0].id} />
      );
      const expectedIcon = iconContainer.querySelector('svg');

      expect(alertIcon).toEqual(expectedIcon);
    });
  });

  describe('Dismissable', () => {
    it('should render a dismissable icon button', () => {
      const { getByLabelText } = renderAlert({ dismissable: true });
      const dismissableIconButton = getByLabelText('Close this message');

      expect(dismissableIconButton).toBeInTheDocument();
    });

    it('should render a dismissable icon button with custom close label text', () => {
      const { getByLabelText } = renderAlert({
        closeLabel: 'Test',
        dismissable: true
      });
      const dismissableIconButton = getByLabelText('Test');

      expect(dismissableIconButton).toBeInTheDocument();
    });

    it('should render a dismissable icon button with the warning variant', () => {
      const { getByLabelText } = renderAlert({
        dismissable: true,
        variant: 'warning'
      });

      const button = getByLabelText('Close this message');
      button.setAttribute('id', 'ignoreButton');
      button.firstChild.setAttribute('id', 'ignoreSvg');
      button.firstChild.setAttribute('aria-labelledby', 'ignoreButton');
      button.firstChild.firstChild.setAttribute('id', 'ignoreTitle');

      expect(button).toMatchSnapshot();
    });

    it('should call passed in onDismiss when dismissable icon button is clicked', () => {
      const onDismissSpy = jest.fn();
      const { getByLabelText } = renderAlert({
        dismissable: true,
        onDismiss: onDismissSpy
      });
      const dismissableIconButton = getByLabelText('Close this message');

      fireEvent.click(dismissableIconButton);

      setTimeout(() => {
        expect(onDismissSpy).toHaveBeenCalled();
      }, 500);
    });
  });

  it('should render custom styles', () => {
    const color = '#cccccc';
    const { container } = renderAlert({ style: { color } });

    expect(container.firstChild).toHaveStyle(`color: ${color}`);
  });

  it('Does not violate accessibility standards', () => {
    const { container } = renderAlert();
    return axe(container.innerHTML).then(result => {
      return expect(result).toHaveNoViolations();
    });
  });
});
