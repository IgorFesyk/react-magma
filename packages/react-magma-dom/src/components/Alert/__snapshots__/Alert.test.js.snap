// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Dismissible should render a dismissible icon button with the warning variant 1`] = `
<button
  aria-label="Close this message"
  class="css-o5nfe3"
  icon="[object Object]"
  id="ignoreButton"
  theme="[object Object]"
>
  <svg
    aria-labelledby="ignoreButton"
    class="icon"
    fill="currentColor"
    height="18"
    id="ignoreSvg"
    viewBox="0 0 28 28"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M27.75,22.5h0L19.25,14l8.49-8.49h0a.75.75,0,0,0,.2-.31.87.87,0,0,0-.2-.93l-4-4a.86.86,0,0,0-.92-.2.91.91,0,0,0-.32.2h0L14,8.75,5.51.26h0a.83.83,0,0,0-.32-.2.86.86,0,0,0-.92.2l-4,4a.88.88,0,0,0,0,1.24h0L8.75,14,.26,22.5h0a.79.79,0,0,0-.2.31.86.86,0,0,0,.2.92l4,4a.88.88,0,0,0,.92.2,1,1,0,0,0,.32-.2h0L14,19.25l8.49,8.49h0a.78.78,0,0,0,.32.2.86.86,0,0,0,.92-.2l4-4a.86.86,0,0,0,.2-.92A.77.77,0,0,0,27.75,22.5Z"
      id="ignoreTitle"
      transform="translate(0 0)"
    />
  </svg>
</button>
`;
